### PR TITLE
Add CMake buildfiles. Use CMake in travis.

### PR DESCRIPTION
**Pull request by eustas**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+add_library(brotli_common STATIC
+  common/dictionary.c)
+add_library(brotli_dec STATIC
+  dec/bit_reader.c
+  dec/decode.c
+  dec/huffman.c
+  dec/state.c)
+add_library(brotli_enc STATIC
+  enc/backward_references.c
```
Comment (eustas) [inline CMakeLists.txt]: Why do not use GLOB? (And have policy that common/, dec/ and enc/ may contain only files to be built)

**Comment by eustas** [comment]
Travis looks happy, indeed.
The only thing left to do is to sign CLA (sorry for the bureaucracy).

File: CMakeLists.txt (hunk 1 above)
Comment (nemequ) [inline CMakeLists.txt]: > Why do not use GLOB? (And have policy that common/, dec/ and enc/ may contain only files to be built)

From the CMake documentation:

> We do not recommend using GLOB to collect a list of source files from your source tree. If no CMakeLists.txt file changes when a source is added or removed then the generated build system cannot know when to ask CMake to regenerate.

I can change it to use glob if you want, but with CI in place I don't think there is much danger of forgetting to add/remove a file, so IMHO it is better to leave as-is.

**Comment by nemequ** [comment]
:o didn't realize you opened a PR for this.  I signed the CLA for a patch to Gipfeli, so that shouldn't be an issue anymore.

Pretty much everything should be working now, the only real exception is TSan, which I don't think is going to happen on Ubuntu 12.04.

The Travis error above looks transient (it happens sometimes); asking Travis to rebuild the commit should work.

**Comment by eustas** [comment]
Don't worry about CLA. Rerunning travis...
Thank you for this PR!

**Comment by eustas** [comment]
It looks like problem with gcc4.5 persists...

**Comment by nemequ** [comment]
That's very odd.  The same commit works for me: https://travis-ci.org/nemequ/brotli/jobs/146810092

**Comment by nemequ** [comment]
Disabled gcc-4.5, everything finally seems to be working.  The failure with ubsan is expected; there is some undefined behavior in brotli (even with `-DBROTLI_BUILD_PORTABLE`).

Still not sure why gcc-4.5 works in my repo but not as a PR, but I guess it's not a big deal.

**Comment by eustas** [comment]
Going to calm down clang-ubsan in few moments.

**Comment by eustas** [comment]
Ubsan is happy now.

**Comment by eustas** [comment]
I signed it!

**Comment by nemequ** [comment]
I signed it!

**Comment by nemequ** [comment]
I'm guessing if **I** opened up a PR it would resolve the CLA thing…  would that help?

**Comment by eustas** [comment]
I think - yes. This bot is always harassing me...

**Comment by nemequ** [comment]
Argh, doesn't look like GitHub will let me file a duplicate PR; it just links me to this one.  I guess if you close this one I could probably do it…

**Comment by eustas** [comment]
Okay, let's try

**Pull request closed without merging**